libcloud/compute/drivers/kamatera.py

```diff
@@ -0,0 +1,639 @@
+        return self.ex_node_operation(node, 'reboot', ex_wait)
+
+    def destroy_node(self, node, ex_wait=True):
+        """
+        Destroy the given node
+
+        :param node:     the node to destroy
+        :type node: :class:`Node`
+
+        :param ex_wait:     wait for destroy to complete (optional)
+        :type ex_wait:      ``bool``
+
+        :rtype: ``bool``
+        """
+        return self.ex_node_operation(node, 'terminate', ex_wait)
+
+    def stop_node(self, node, ex_wait=True):
+        """
+        Stop the given node
+
+        :param node:     the node to stop
+        :type node: :class:`Node`
+
+        :param ex_wait:     wait for stop to complete (optional)
+        :type ex_wait:      ``bool``
+
+        :rtype: ``bool``
+        """
+        return self.ex_node_operation(node, 'poweroff', ex_wait)
+
+    def start_node(self, node, ex_wait=True):
+        """
+        Start the given node
+
+        :param node:     the node to start
+        :type node: :class:`Node`
+
+        :param ex_wait:     wait for start to complete (optional)
+        :type ex_wait:      ``bool``
+
+        :rtype: ``bool``
+        """
+        return self.ex_node_operation(node, 'poweron', ex_wait)
+
+    def ex_node_operation(self, node, operation, wait=True):
+        """
+        Run custom operations on the node
+
+        :param node:     the node to run operation on
+        :type node: :class:`Node`
+
+        :param operation:   the operation to run
+        :type operation:   ``str``
+
+        :param ex_wait:     wait for destroy to complete (optional)
+        :type ex_wait:      ``bool``
+
+        :rtype: ``bool``
+        """
+        if node.id:
+            request_data = {'id': node.id}
+        elif node.name:
+            request_data = {'name': node.name}
+        else:
+            raise ValueError('Invalid node for %s node operation: '
+                             'missing id / name' % operation)
+        if operation == 'terminate':
+            request_data['force'] = True
+        command_id = self.connection.request(
+            '/service/server/%s' % operation, method='POST',
+            data=json.dumps(request_data)).object[0]
+        if wait:
+            self.ex_wait_command(command_id)
+        else:
+            node.extra['%s_command_id' % operation] = command_id
+        return True
+
+    def ex_get_location(self, id, name=None, country=None):
+        """
+        Get a NodeLocation object to use for other methods
+
+        :param id:     Location ID - uppercase letters code (required)
+        :type id:      ``str``
+
+        :param name:     Location Name (optional)
+        :type name:      ``str``
+
+        :param name:     Location country (optional)
+        :type name:      ``str``
+
+        :rtype: :class:`.NodeLocation`
+        """
+        return NodeLocation(id=id, name=name, country=country, driver=self)
+
+    def ex_get_size(self, ramMB, diskSizeGB, cpuType, cpuCores,
```

Comment:
We tend to use underscore separated variable names, but since this is an extension method, this notation is also acceptable if it matches the API parameters.